org.mwc.debrief.lite/src/main/java/org/mwc/debrief/lite/menu/DebriefRibbonTimeController.java

```diff
@@ -1114,28 +1118,35 @@ public static void setButtonsEnabled(List<Command> commands,
   public static void updatePlayBtnUI(
       final AbstractCommandButton playCommandButton, final boolean isPlaying)
   {
-    final String image;
-    if (isPlaying)
-      image = PLAY_IMAGE;
-    else
-      image = STOP_IMAGE;
-
-    final String tooltip = isPlaying ? STOP_TEXT : START_TEXT;
-
-    final RichTooltip.Builder builder = RichTooltip.builder();
-    final RichTooltip richTooltip = builder.setTitle("Timer")
-        .addDescriptionSection(tooltip).build();
-    playCommandButton.setActionRichTooltip(richTooltip);
-
-    // switch the icon
-    final Image playStopinImage = MenuUtils.createImage(image);
-    final ImageWrapperResizableIcon imageIcon = ImageWrapperResizableIcon
-        .getIcon(playStopinImage, MenuUtils.ICON_SIZE_16);
-
-    playCommandButton.setExtraText(tooltip);
-//    playCommandButton.setIcon(imageIcon);
-    playCommand.setIconFactory(ResizableIconFactory.factory(imageIcon));
-    playCommandButton.repaint();
+    
+      final String image;
+      if (isPlaying)
+        image = PLAY_IMAGE;
+      else
+        image = STOP_IMAGE;
+
+      final String tooltip = isPlaying ? STOP_TEXT : START_TEXT;
+
+      final RichTooltip.Builder builder = RichTooltip.builder();
+      final RichTooltip richTooltip = builder.setTitle("Timer")
+          .addDescriptionSection(tooltip).build();
+      playCommandButton.setActionRichTooltip(richTooltip);
+      SwingUtilities.invokeLater(()-> {
+        // switch the icon
+        final Image playStopinImage = MenuUtils.createImage(image);
+        final ImageWrapperResizableIcon imageIcon = ImageWrapperResizableIcon
+            .getIcon(playStopinImage, MenuUtils.ICON_SIZE_16);
+  
+        playCommandButton.setExtraText(tooltip);
+       // playCommand.setIconFactory(ResizableIconFactory.factory(imageIcon));
+        
+        playCommandButton.setIcon(imageIcon);
+        if(playButton!=playCommandButton)
```

Comment:
I don't quite understand why we're comparing buttons here, but it does look like a symptom of all the static public objects - where we can't keep track of what we're looking at.  IMHO...

Reply:
`playButton` I keep ref soon as user hit play, so i found some case button u have in variable and button passed to method not same so in case it is not same I set icon as one I collect at event is the one in UI.